packages/datastore/datastore/src/replication/mutations/MutationsQueue.ts

```diff
@@ -57,124 +108,155 @@ export class MutationsReplicationQueue {
         this.open = false;
       }
     });
+
+    // Intentionally async to start replication in background
+    this.options.networkIndicator.isNetworkReachable().then((result) => {
+      if (this.open === result) {
+        // No state change
+        return;
+      }
+
+      if (result === true) {
+        // Going online
+        this.process();
+      }
+      this.open = result;
+    });
   }
 
   /**
-   * Perform mutation request that will:
-   *
-   * - Add request to the array of the current requests
-   * - Save **all** requests to the store for this model
-   * - Process requests (subject to network availability)
-   * - Map id's
-   *
-   * @param request
+   * Save user change to bereplicated by engine
    */
-  public addMutationRequest(request: MutationRequest) {
-    this.items.push(request);
-    // Save entire queue
-    this.persistQueueTo();
+  public async saveChangeForReplication(model: Model, data: any, eventType: CRUDEvents, store: LocalStorage) {
+    // Actual graphql queries need to be persisted at the time of request creation
+    // This will ensure consistency for situations when model changed (without migrating queue)
+    const storeName = model.getStoreName();
+    const operations = this.modelMap[storeName];
+    invariant(operations, "Missing GraphQL mutations for replication");
+
+    const mutationRequest = {
+      storeName,
+      data,
+      eventType
+    };
+    await this.enqueueRequest(mutationRequest);
```

Comment:
It seems like provision has made here to include the enqueue operation in the CRUD transaction. But the passed store object is not actually used. If this is a TODO or a mistake, I can fix it for you

Reply:
mistake. This might fix tests